components/ILIAS/Mail/src/Cron/ScheduledMailsCron.php

```diff
@@ -0,0 +1,209 @@
+    }
+
+    public function getDescription(): string
+    {
+        $this->init();
+
+        return $this->lng->txt('mail_cron_scheduled_mails_desc');
+    }
+
+    public function hasAutoActivation(): bool
+    {
+        return true;
+    }
+
+    public function hasFlexibleSchedule(): bool
+    {
+        return true;
+    }
+
+    public function getDefaultScheduleType(): JobScheduleType
+    {
+        return JobScheduleType::DAILY;
+    }
+
+    public function getDefaultScheduleValue(): ?int
+    {
+        return 1;
+    }
+
+    public function run(): JobResult
+    {
+        $this->init();
+
+        $job_result = new JobResult();
+        $job_result->setStatus(JobResult::STATUS_OK);
+
+        ilLoggerFactory::getLogger('mail')->info('Start sending scheduled mails from all users.');
+
+        $mails = $this->getOutboxMails();
+        $sent_mail_ids = [];
+        foreach ($mails as $mail) {
+            try {
+                $mailer = $this->umail
+                    ->withContextId(ilContext::CONTEXT_CRON);
+
+                $mailer->setSaveInSentbox(true);
+
+                $mailer->autoresponder()->enableAutoresponder();
+                $errors = $mailer->enqueue(
+                    $mail['rcp_to'],
+                    $mail['rcp_cc'],
+                    $mail['rcp_bcc'],
+                    $mail['m_subject'],
+                    $mail['m_message'],
+                    unserialize($mail['attachments'], ['allowed_classes' => false]),
+                    (bool) ($mail['use_placeholders'] ?? false)
+                );
+
+                if (empty($errors)) {
+                    $sent_mail_ids[] = (int) ($mail['mail_id'] ?? 0);
+                }
+            } catch (Throwable $e) {
+                $job_result->setStatus(JobResult::STATUS_FAIL);
+                ilLoggerFactory::getLogger('mail')->error(
+                    'Error sending scheduled mail with id ' . ((string) ($mail['mail_id'] ?? 'unknown')) . ': ' .
+                    $e->getMessage() . '\n' . $e->getTraceAsString()
+                );
+                $job_result->setMessage(substr($e->getMessage() . ' ' . $e->getTraceAsString(), 0, 4000));
+
+                return $job_result;
+            }
+        }
+        $this->mail->deleteMails($sent_mail_ids);
+        ilLoggerFactory::getLogger('mail')->info(
+            'Sent ' . count($sent_mail_ids) . ' scheduled mails and removed them from outbox.'
+        );
+        $job_result->setMessage('Processed ' . count($sent_mail_ids) . ' mails.');
+
+        return $job_result;
+    }
+
+    /**
+     * @return Generator<list{
+     *     mail_id: int,
+     *     rcp_to: string,
+     *     rcp_cc: string,
+     *     rcp_bcc: string,
+     *     m_subject: string,
+     *     m_message: string,
+     *     attachments: string,
+     *     use_placeholders: bool,
+     *     schedule_datetime: string,
+     *     schedule_timezone: string
+     * }>
```

Comment:
Maybe we could enhance `MailDeliveryData` (by introducing the `scheldue_datetime` (without additional timezone) and use an object of this class instead?